crates/ruma-common/src/events/room/message.rs

```diff
@@ -200,6 +200,131 @@ impl RoomMessageEventContent {
         }
     }
 
+    /// Create a new reply with the given message.
+    ///
+    /// If `forward_thread` is `true`, the [`Relation::Thread`] in `original_message` is forwarded
+    /// if it exists. This should be set if your client doesn't support threads (see [MSC3440]).
+    ///
+    /// If `forward_thread` is `false`, this creates a reply in the main conversation even if the
+    /// original message is in a thread. This should be used if you client supports threads and you
+    /// explicitly want that behavior.
+    ///
+    /// If `message` is a text or notice message, it is modified to include the rich reply fallback.
+    ///
+    /// [MSC3440]: https://github.com/matrix-org/matrix-spec-proposals/pull/3440
+    #[cfg(feature = "unstable-msc3440")]
+    pub fn reply(
+        message: MessageType,
+        original_message: &OriginalRoomMessageEvent,
+        forward_thread: bool,
```

Comment:
This seems like the type of parameter you would pass inline, instead of having it in a local variable first. As such, I think we use a dedicated enum instead of a `bool` here.